distributed/shuffle/_buffer.py

```diff
@@ -45,6 +45,7 @@ class ShardsBuffer(Generic[ShardType]):
 
     shards: defaultdict[str, _List[ShardType]]
     sizes: defaultdict[str, int]
+    sizes_detail: defaultdict[str, _List[int]]
```

Comment:
I blindly copy-pasted `shards`. Why does that have to be a `_List`?

Reply:
That was done to trick our serialization stack into _not_ trying to be fancy and traverse into the list when serializing before sending the data via comms.

Reply:
list prohibits the protocol to iterate into the collection. If the list is very long, this can be expensive. On the flipside, if we're not walking down, the entire thing is pickled

Reply:
changed to regular list